packages/components/src/Typeahead/Typeahead.component.renderers.js

```diff
@@ -35,6 +36,7 @@ export function renderInputComponent(props) {
 			<ControlLabel srOnly htmlFor={key}>
 				Search
 			</ControlLabel>
+			{get(props, 'selecteditem.icon') && <Icon {...props.selecteditem.icon} className={theme['item-icon']} />}
```

Comment:
inverting className props and the spreading would alllow className customization through props